test/support/fixture_helper.rb

```diff
@@ -0,0 +1,133 @@
+# typed: ignore
+# frozen_string_literal: true
+
+require "rails_test_helper"
+
+module FixtureHelper
+  TEMP_FIXTURE_DIR = ROOT.join("tmp", "fixtures").to_s
+  FileUtils.mkdir_p(TEMP_FIXTURE_DIR)
+
+  Minitest.after_run do
+    # This does not get called on early exit, e.g. `Ctrl-C` or `(byebug) exit`.
+    FileUtils.remove_entry(TEMP_FIXTURE_DIR, true)
+  end
+
+  def around
+    old_working_dir = Dir.pwd
+    cache_rails_paths
+
+    yield
+
+    restore_rails_paths
+    Dir.chdir(old_working_dir)
+    FileUtils.remove_entry(@app_dir, true) if defined? @app_dir
+  end
+
+  def copy_template(template)
+    copy_dir("test/fixtures/#{template}")
+    Dir.chdir(app_dir)
+
+    case template
+    when :minimal
+      rails_autoload_paths(paths_to("/components/sales/app/models"))
+      rails_eager_load_paths([])
+      rails_autoload_once_paths([])
+      rails_root_path(app_dir)
+    when :skeleton
+      rails_autoload_paths(paths_to("components/sales/app/models"))
+      rails_eager_load_paths(paths_to("components/platform/app/models"))
+      rails_autoload_once_paths(
+        paths_to(
+          "components/timeline/app/models",
+          "components/timeline/app/models/concerns",
+          "vendor/cache/gems/example/models",
+        )
+      )
+      rails_root_path(app_dir)
+      # make sure PrivateThing.constantize succeeds to pass the privacy validity check
+      require "fixtures/skeleton/components/timeline/app/models/private_thing.rb"
+    else
+      raise "Unknown fixture template #{template}"
+    end
+  end
+
+  def copy_dir(path)
+    root = Dir.mktmpdir(name, TEMP_FIXTURE_DIR)
+    FileUtils.cp_r("#{path}/.", root)
+    @app_dir = root
+  end
+
+  def config
+    @config ||= Packwerk::Configuration.from_path(app_dir)
+  end
+
+  def merge_into_yaml_file(relative_path, hash)
```

Comment:
Either create a new yaml file at the indicated path, or update the existing one by treating its contents as a hash and merging in the new hash.

If there is an existing array value, e.g.
```yml
# path/to/file.yml
a: some_value
b:
- 1
- 2
c:
- hi
```
Then calling 
```ruby
merge_into_yaml_file("path/to/file.yml", { b: [3,4], c: "bye" })
```
Will result in this yml
```yml
# path/to/file.yml
a: some_value
b:
- 1
- 2
- 3
- 4
c:
- hi
- bye
```